conf/ux/expectations.conf.schema.json

```diff
@@ -0,0 +1,220 @@
+        "modes": {
+            "description": "The various modes we can be in and their specifications. The active mode is the first mode in this list with a schedule matching the current date or with no schedule.",
+            "type": "array",
+            "items": {
+                "$ref": "#/definitions/modeType"
+            },
+            "contains": {
+                "$comment": "Must contain a \"default\" element -- one with no schedule",
+                "not": {"required": ["schedule"]}
+            }
+        }
+    },
+    "required": ["modes"],
+    "definitions": {
+        "modeType": {
+            "description": "Specifies a collection mode that defines expectation and notification settings for a given time period",
+            "type": "object",
+            "properties": {
+                "enabled": {
+                    "description": "Whether or not to enable this mode; mode is enabled if this property is omitted",
+                    "type": "boolean"
+                },
+                "label": {
+                    "description": "A human-readable label for the mode",
+                    "type": "string"
+                },
+                "confidenceThreshold": {
+                    "description": "Only display yellow labels with confidence greater than this threshold (0 to display all, 1 to display none)",
+                    "type": "number",
+                    "minimum": 0,
+                    "maximum": 1
+                },
+                "schedule": {
+                    "$ref": "#/definitions/scheduleType"
+                },
+                "rules": {
+                    "description": "A list of rules that define the expectation and notification settings for label types",
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/ruleType"
+                    },
+                    "allOf": [
+                        {
+                            "contains": {
+                                "$comment": "Must contain a rule for red labels",
+                                "properties": {"trigger": {"const": -1}}
+                            }
+                        },
+                        {
+                            "contains": {
+                                "$comment": "Must contain a rule for all yellow labels",
+                                "properties": {"trigger": {"const": 1}}
+                            }
+                        }
+                    ]
+                }
+            },
+            "required": ["label", "confidenceThreshold", "rules"]
+        },
+        "scheduleType": {
+            "description": "Describes when the mode is active; omit to make always active",
+            "type": "object",
+            "properties": {
+                "startDate": {
+                    "description": "The base date from which to measure time; omit time zone to signify user's local time",
```

Comment:
The logic for this part is actually relatively simple; the phone's time zone is embedded in the trip data object, so the server just inserts it if we're missing a time zone. (The complicated part is the schedule matching, regardless of time zones, and in particular figuring out how many months/weeks/days there are between two dates — see `expectation_notification_config.py:64`).

Let's say we want to do primary data collection for one week, Monday to Sunday, every eight weeks. For simplicity, say we want the user to label all trips at the end of the day under primary data collection and none under secondary data collection. Under the existing schema, we can implement this as a schedule that begins at `00:00` user's local time and lasts `7` days, and specify a `startDate` that is a Monday. Then the user will have trips appear on their `To Label` screen for exactly 7 days every 56 days; if they label trips at the end of each day as they are supposed to, they will be interacting with the app 1 in 8 days.

Say we remove the ability to specify a time in the user's local time zone. We implement the schedule as beginning at `00:00` UTC. If the user is in a US time zone, primary mode will begin on Sunday evening and last until the next Sunday evening. Though this is the same length of time and on average will cover the same number of trips, it now spans 8 days instead of 7. If the user is labeling at the end of each day, they will be interacting with the app 1 in 7 days. This is worse.

To summarize, the ability to use a user's local time zone minimizes the number of days the user has to interact with the app. Even if they'll be labeling the same number of trips either way, every day we can avoid making the user do something is a hassle avoided.

I suppose if we wanted to simplify we could _only_ use local time and not give the option to specify a time zone.

Reply:
Again, let's discuss this after tomorrow. Right now, I'm going to merge this and get the actual recommendations to show up.